tests/envs/MaMuJoCo/test_MaMuJoCo.py

```diff
@@ -0,0 +1,498 @@
+from __future__ import annotations
+
+import collections
+
+import numpy as np
+import pytest
+from multiagent_mujoco.mujoco_multi import MultiAgentMujocoEnv
+from pettingzoo.test import parallel_api_test
+
+scenario_conf = collections.namedtuple("scenario_conf", "scenario, conf")
+
+pre_defined_factorizations = [
+    scenario_conf("InvertedPendulum", None),  # For Debugging
+    scenario_conf("Ant", None),
+    scenario_conf("Ant", "2x4"),
+    scenario_conf("Ant", "2x4d"),
+    scenario_conf("Ant", "4x2"),
+    scenario_conf("HalfCheetah", "2x3"),
+    scenario_conf("HalfCheetah", "6x1"),
+    scenario_conf("HalfCheetah", None),
+    scenario_conf("Hopper", "3x1"),
+    scenario_conf("Hopper", None),
+    scenario_conf("Humanoid", "9|8"),
+    scenario_conf("Humanoid", None),
+    scenario_conf("HumanoidStandup", "9|8"),
+    scenario_conf("HumanoidStandup", None),
+    scenario_conf("Reacher", "2x1"),
+    scenario_conf("Reacher", None),
+    scenario_conf("Swimmer", "2x1"),
+    scenario_conf("Swimmer", None),
+    scenario_conf("Pusher", "3p"),
+    scenario_conf("Pusher", None),
+    scenario_conf("Walker2d", "2x3"),
+    scenario_conf("Walker2d", None),
+]
+
+sample_configurations = [
+    scenario_conf("manyagent_swimmer", "10x2"),
+    # scenario_conf("manyagent_swimmer", "5x4"),
+    scenario_conf("manyagent_swimmer", "6x1"),
+    scenario_conf("manyagent_ant", "2x3"),
+    scenario_conf("manyagent_ant", "3x1"),
+    scenario_conf("coupled_half_cheetah", "1p1"),
+    scenario_conf("coupled_half_cheetah", None),
+]
+
+observation_depths = [None, 0, 1, 2]
+
+
+def assert_dict_numpy_are_equal(
+    dict_a: dict[any, np.ndarray], dict_b: dict[any, np.ndarray]
+) -> None:
+    """Asserts that 2 dictionaries have the key/value pairs (given that the values are `np.ndarray`)."""
+    assert dict_a.keys() == dict_b.keys()
+    for key in dict_a.keys():
+        assert len(dict_a[key]) == len(dict_b[key])
+        assert (dict_a[key] == dict_b[key]).all()
+
+
+@pytest.mark.parametrize("observation_depth", observation_depths)
+@pytest.mark.parametrize("task", pre_defined_factorizations + sample_configurations)
+def test_general(observation_depth, task) -> None:
+    """Asserts that the environments are compliant with `pettingzoo.utils.env.ParallelEnv` API."""
+    parallel_api_test(
+        MultiAgentMujocoEnv(task.scenario, task.conf, agent_obsk=observation_depth),
+        num_cycles=1_000_000,
```

Comment:
This is alot of cycles, is this normal? How long does it take to run for each environment locally?

Reply:
It takes ~34sec on my laptop for the all the environments